### ISSUE TITLE
Igniters could provide more fully-fleshed email sender modules

### ISSUE THREAD

**Issue by sevenseacat** [team-alembic/ash_authentication#815]
Igniters such as `ash_authentication.add_strategy password` generate sender modules that look like this:

```elixir
defmodule Tunez.Accounts.User.Senders.SendPasswordResetEmail do
  @moduledoc """
  Sends a password reset email
  """

  use AshAuthentication.Sender
  use TunezWeb, :verified_routes

  @impl true
  def send(_user, token, _) do
    # Example of how you might send this email
    # Tunez.Accounts.Emails.send_password_reset_email(
    #   user,
    #   token
    # )

    IO.puts("""
    Click this link to reset your password:

    #{url(~p"/password-reset/#{token}")}
    """)
  end
end
```

It took me a bit to realize that to actually send the email, I would have to create the `Tunez.Accounts.Emails` module, and define that function with the right copy and then create and send an email.

My thinking was that the igniter could generate something more fully-featured, like the example in the tutorial - https://hexdocs.pm/ash_authentication/confirmation.html#confirming-newly-registered-users, eg.

* Create/update an Emails module with the new function eg. `send_password_reset_email` (and a `deliver` function if one doesn't exist)
* Update the sender module to call that function in the Emails module
* The email content can then be moved to the Emails module function
* If adding the `deliver` function, it can print out the content and perhaps have the other content from the docs included but commented out
```elixir
  # For simplicity, this module simply logs messages to the terminal.
  # You should replace it by a proper email or notification tool, such as:
  #
  #   * Swoosh - https://hexdocs.pm/swoosh
  #   * Bamboo - https://hexdocs.pm/bamboo
  #
  defp deliver(to, subject, body) do
    IO.puts("Sending email to #{to} with subject #{subject} and body #{body}")

    new()
    |> from({"Zach", "zach@ash-hq.org"}) # TODO: Replace with your email
    |> to(to_string(to))
    |> subject(subject)
    |> put_provider_option(:track_links, "None")
    |> html_body(body)
    |> MyApp.Mailer.deliver!()
  end
```

**Comment by zachdaniel** [comment]
So, we can do this, but to do it right we likely need to add a couple swoosh specific helpers, i.e `Igniter.Libs.Swoosh`. Eventually we'd like these to be extracted out to their respective libraries.

**Comment by zachdaniel** [comment]
@sevenseacat I imagine you'd like this in for the auth chapter in the book. It also makes sense from our live stream today with Peter who expected to go to the dev mailbox and see the email right away. We ought to be able to set it up without too much hassle.

**Comment by zachdaniel** [comment]
I'll try and do this tomorrow, but it might have to wait til next week.

**Comment by sevenseacat** [comment]
Yes I did the exact same thing as Peter; when I saw the output in my logs I expected to also see it in an email in the dev mailbox. 

Maybe it doesn't have to go that far, but having the Emails module set up ready to plug a mailer straight into would be a good compromise.

**Comment by jimsynz** [comment]
Closed by #835
